fmt/src/header/parser.rs

```diff
@@ -71,29 +73,33 @@ pub fn parse_header(
     };
 
     // 2. has header
-    let existing_header = {
+    let mut existing_header = || -> bool {
         // skip blank lines
-        while let Some(l) = line.as_ref()
-            && l.trim().is_empty()
-        {
+        while let Some(l) = line.as_ref() {
+            if !l.trim().is_empty() {
+                break;
+            }
             line = file_content.next_line();
         }
 
         // check if there is already a header
         let mut got_header = false;
-        if let Some(l) = line.as_ref()
-            && header_def.is_first_header_line(l)
-        {
+        if let Some(l) = line.as_ref() {
+            if header_def.is_first_header_line(l) {
+                return false;
```

Comment:
OK. I see that return is helpful to ensure short circuit logic. You may try to extract this block into a function, rather than keep it a closure, if you would go this way.